mountpoint-s3/src/inode/readdir.rs

```diff
@@ -186,6 +186,11 @@ impl ReaddirHandle {
         self.inner.update_from_remote(self.dir_ino, entry.name(), remote_lookup)
     }
 
+    pub async fn rewind(&self) {
+        self.readded.lock().unwrap().take();
+        self.iter.lock().await.rewind();
+    }
```

Comment:
Instead of having to reason about this stateful thing, can we just recreate the whole `ReaddirHandle` by calling `self.superblock.readdir` again? It's a little bit wasteful because it needs to recreate the local entries and the ListObjects stream, but it's also much easier to get correct. 

We can go do something fancier if it turns out we don't like the performance of the simpler version. But "something fancy" might be as simple as just keeping the `last_response` code around.

Reply:
I'll give this a try!

Reply:
Changed to replacing the handled.